src/include/daos_prop.h

```diff
@@ -439,6 +448,19 @@ int
 daos_prop_entry_dup_co_roots(struct daos_prop_entry *dst,
 			     struct daos_prop_entry *src);
 
+/**
+ * Check if DAOS (pool or container property) label string is valid
+ * (string length <= DAOS_PROP_LABEL_MAX_LEN, and conforms to
+ *  allowed characters defined by DAOS_LABEL_REGEX).
+ *
+ * \param[in]	label	Label string
+ *
+ * \return		true		Label meets length/format requirements
+ *			false		Label is not valid length or format
+ */
+bool
+daos_label_is_valid(const char *label);
```

Comment:
All things being equal it's better if we can pass in a size_t along with char * strings to minimise the use of strnlen().  We have a significant number of public interfaces that don't do this however so we may decide that that boat has sailed.  At least for container labels we know the max length so we can use strnlen() rather than strlen().